manifests/init.pp

```diff
@@ -292,10 +292,30 @@
   if $pulpcore and !$pulpcore_mirror {
     include foreman
 
+    if $pulp {
+      include katello::pulp
+      include pulpcore::plugin::migration
+      $mongo_db_config = {
+        'mongo_db_name'         => $katello::pulp::mongodb_name,
+        'mongo_db_seeds'        => $katello::pulp::mongodb_seeds,
+        'mongo_db_username'     => $katello::pulp::mongodb_username,
+        'mongo_db_password'     => $katello::pulp::mongodb_password,
+        'mongo_db_replica_set'  => $katello::pulp::mongodb_replica_set,
+        'mongo_db_ssl'          => $katello::pulp::mongodb_ssl,
+        'mongo_db_ssl_keyfile'  => $katello::pulp::mongodb_ssl_keyfile,
+        'mongo_db_ssl_certfile' => $katello::pulp::mongodb_ssl_certfile,
+        'mongo_db_verify_ssl'   => $katello::pulp::mongodb_verify_ssl,
+        'mongo_db_ca_path'      => $katello::pulp::mongodb_ca_path,
+      }
+    } else {
+      $mongo_db_config = {}
+    }
+
     class { 'pulpcore':
       remote_user_environ_name => 'HTTP_REMOTE_USER',
       manage_apache            => false,
       servername               => $foreman::servername,
+      *                        => $mongo_db_config,
```

Comment:
I do see the value that @ekohl mentioned with this set of parameters living on the `migration` class and declaring that directly rather than as an includes. It encapsulates the migration portion a bit cleaner. What do you think @wbclark ?

Reply:
I agree, there is no reason to have the parameters outside of the `migration` plugin. I've opened https://github.com/theforeman/puppet-pulpcore/pull/63 to move them and will push a new version of this commit which respects that change.

Reply:
@ehelms I've updated this for https://github.com/theforeman/puppet-pulpcore/pull/63